hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,264 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "edn",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  bus_host: "none",
+  param_list: [
+    { name: "NumEndPoints",
+      desc: "Number of end points to be supported by this block",
+      type: "int",
+      default: "4",
+      local: "true"
+    },
+    { name: "EndPointBusWidth0",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 0",
+      type: "int",
+      default: "64",
+      local: "true"
+    },
+    { name: "EndPointBusWidth1",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 1",
+      type: "int",
+      default: "32",
+      local: "true"
+    },
+    { name: "EndPointBusWidth2",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 2",
+      type: "int",
+      default: "16",
+      local: "true"
+    },
+    { name: "EndPointBusWidth3",
+      desc: "Power of two width of the obsfucated entropy data bus for end point 3",
+      type: "int",
+      default: "8",
+      local: "true"
+    },
+  ],
+  interrupt_list: [
+    { name: "edn_cmd_req_done"
+      desc: "Asserted when a software csrng request has completed."}
+    { name: "edn_fifo_err"
+      desc: "Asserted when a FIFO error occurs."}
+  ],
+  inter_signal_list: [
+    { struct:  "edn",  // edn_req_t, edn_rsp_t
+      type:    "req_rsp",
+      name:    "edn_cmd",
+      act:     "rsp",
+      width:   "4",
+      package: "edn_pkg",
+    }
+  ],
+
+  regwidth: "32",
+  registers: [
+    { name: "REGEN",
+      desc: "Register write enable for all control registers",
+      swaccess: "rw1c",
+      hwaccess: "hro",
+      fields: [
+        {
+            bits:   "0",
+            desc: ''' When true, the EDN_CONF, EDN_RCT_HEALTH, EDN_APT_HEALTH and EDN_SEED registers can be modified.
+            When false, they become read-only. Defaults true, write one to clear. Note that this needs to be
+            cleared after initial configuration at boot in order to lock in the listed register settings.
+            '''
+            resval: 1
+        },
+      ]
+    },
+    { name: "CTRL",
+      desc: "EDN control register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [ 
+        {
+            bits: "0",
+            name: "EDN_ENABLE",
+            desc: '''
+                  Setting this bit will enable the EDS module. 
+                  '''
+        },
+        {
+            bits: "1",
+            name: "BCAST_MODE_DIS",
+            desc: '''
+                  Setting this bit will disable the EDN block to self ack the ENDPOINT0 entropy request
+                  bus. This mode for ENDPOINT0 will ignore the request input.
```

Comment:
I think this should probably be something like "If this bit is set, the EDN block will not run in broadcast mode for endpoint 0." The behaviour of the ACK signal in broadcast mode probably belongs somewhere else.

Also, all the negatives are a bit confusing: is there a strong reason not to have a bit that enables the mode instead? (And similarly for BOOT_REQ_DIS below)

Reply:
Rewrote this to be more clear.
The negatives are a bit confusing - they used to be positive. Because of the recent boot mode feature, a single reg write is desired to start all modules. So if the default is zero, then this function will be enabled. At least that is the reason.

Reply:
Oh, I see. I don't think I realised we preferred registers that reset to zero. In which case, I agree this makes sense.

Reply:
Ugh.   Because of "amends" I can no longer figure out if this is resolved or not, nor can I find this text.   Resolution unclear?